### PR TITLE
fix(pr-review-loop): stop while Codex review is pending

### PR DESCRIPTION
**Pull request by lhpaul**
## Summary

- Surface Codex final no-response states as `RESULT=waiting_on_reviewer` with current-head trigger telemetry instead of reviewer-unavailable escalation.
- Propagate the new result through `pr-review-loop.sh`, summary rendering, compare verdicts, and exit code 4.
- Document the Protocol 91 handling and add focused tests for pending triggers, duplicate-trigger avoidance, stale evidence, and wrapper propagation.

Closes #1603

## Validation

- `bash -n scripts/development-workflow/pr-review-loop.sh scripts/development-workflow/codex-github-reviewer.sh scripts/development-workflow/tests/test-pr-review-loop.sh`
- `bash scripts/development-workflow/tests/test-pr-review-loop.sh --area 1 --area 12 --area 13` (448 passed, 0 failed)
- `shellcheck --severity=warning scripts/development-workflow/pr-review-loop.sh scripts/development-workflow/codex-github-reviewer.sh scripts/development-workflow/tests/test-pr-review-loop.sh`
- `python3 scripts/lint/workflow-shell-guard-lint.py --base-ref origin/develop`
- `bash scripts/lint/check-changelog-duplicate-headers.sh CHANGELOG.md`
- `npx markdownlint-cli2 "CHANGELOG.md" "docs/workflow/development-workflow/protocols/91-orchestrate-work-protocol.md"`
- `git diff --check`

---

> [!NOTE]
> **Medium Risk**
> Changes automated PR review orchestration and exit semantics for Codex; misclassification could block merges or spam triggers, but behavior is fail-closed (wait vs escalate) with broad test coverage.
> 
> **Overview**
> When **Codex GitHub review** exhausts its wait budget but a **current-head trigger is already posted** and no terminal bot evidence exists, the reviewer script now returns **`WAITING_ON_REVIEWER` (exit 4)** with trigger/head telemetry (`REASON=codex-github-review-pending`, `PENDING_REVIEW_*`) instead of **`TIMED_OUT` (exit 2)** and reviewer-unavailable escalation. Exit code 2 is narrowed to API/auth/setup failures while waiting.
> 
> **`pr-review-loop.sh`** propagates **`RESULT=waiting_on_reviewer`**, maps it to compare-mode **`waiting`**, posts an appropriate loop summary, and exits **4**. Pending state does **not** dispatch fixers, count as a required-reviewer failure, or escalate on ledger persist failure.
> 
> **Protocol 91** documents orchestrator behavior: stop the local runner, avoid duplicate triggers and readiness/merge steps, and re-run Step 7 when terminal evidence arrives. The readiness gate treats **`waiting_on_reviewer`** like other non-clean loop results.
> 
> Tests cover idempotent pending triggers, wrapper propagation, and scenarios that previously expected timeout/unavailable now expecting **waiting**.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 6d75f4833e6d36b6653507d00f20e748c027c22b. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by lhpaul** [comment]
### Step 7a Internal Review Gate Summary

**Result:** clean
**Reviewer:** Codex native review
**Scope:** #1603 implementation on `fix/1603-codex-pending-review-stop` targeting `develop`

Reviewed the implementation diff for the Codex pending-review terminal state, including helper exit-code mapping, outer reviewer-loop aggregation, fail-closed readiness docs, changelog entry, and focused tests. No blocking issues found.

**Validation evidence:**
- `bash -n scripts/development-workflow/pr-review-loop.sh scripts/development-workflow/codex-github-reviewer.sh scripts/development-workflow/tests/test-pr-review-loop.sh`
- `bash scripts/development-workflow/tests/test-pr-review-loop.sh --area 1 --area 12 --area 13` (448 passed, 0 failed)
- `shellcheck --severity=warning scripts/development-workflow/pr-review-loop.sh scripts/development-workflow/codex-github-reviewer.sh scripts/development-workflow/tests/test-pr-review-loop.sh`
- `python3 scripts/lint/workflow-shell-guard-lint.py --base-ref origin/develop`
- `bash scripts/lint/check-changelog-duplicate-headers.sh CHANGELOG.md`
- `npx markdownlint-cli2 "CHANGELOG.md" "docs/workflow/development-workflow/protocols/91-orchestrate-work-protocol.md"`
- `git diff --check`

**Comment by lhpaul** [comment]
bugbot run

**Comment by lhpaul** [comment]
### Automated Reviewer Loop Summary

**Result:** clean — no blocking findings
**Platforms:** pr-agent (clean), bugbot (clean)
**Findings:** 0 blocking, 0 suggestions


**Ready reviewer phase:** No net-new blocker was found after the draft GitHub gate.
**Ready-phase platforms:** bugbot

**Advisory findings (non-blocking):**
- Possible Issue ([view comment](https://github.com/lhpaul/ai-dev-framework-template/pull/1614#issuecomment-5425833466))
- Possible Issue ([view comment](https://github.com/lhpaul/ai-dev-framework-template/pull/1614#issuecomment-5425833466))

*Posted automatically by `pr-review-loop.sh`.*

<details>
<summary>Reviewer-loop history (2 iterations)</summary>


```json
{
  "schema": "reviewer_loop_history.v1",
  "pr_number": 1614,
  "updated_at": "2026-08-26T13:24:33Z",
  "history_status": "available",
  "history_unavailable_reason": "",
  "entries": [
    {
      "iteration": 1,
      "recorded_at": "2026-08-26T13:18:52Z",
      "head_sha": "567afe8aa6fa4ad766d07fd9a3ae6da43e146d20",
      "run_id": "run-items-1603-20260826T131415Z",
      "result": "needs_fixes",
      "reason": "unresolved_review_threads",
      "platforms": [
        "pr-agent (clean)",
        "bugbot (clean)"
      ],
      "blocking_count": 1,
      "suggestion_count": 0,
      "unresolved_thread_count": 1,
      "late_threads_found": 0,
      "phase_after_clean": {
        "enabled": true,
        "platforms": [
          "bugbot"
        ],
        "started": true,
        "net_new_blocker": true,
        "blocking_platform": "review_threads"
      },
      "small_findings_only": false,
      "small_findings_stop": false,
      "small_findings_rounds": 0,
      "small_findings_required_rounds": 2,
      "small_findings_paths": []
    },
    {
      "iteration": 2,
      "recorded_at": "2026-08-26T13:24:33Z",
      "head_sha": "6d75f4833e6d36b6653507d00f20e748c027c22b",
      "run_id": "run-items-1603-20260826T132056Z",
      "result": "clean",
      "reason": "",
      "platforms": [
        "pr-agent (clean)",
        "bugbot (clean)"
      ],
      "blocking_count": 0,
      "suggestion_count": 0,
      "unresolved_thread_count": 0,
      "late_threads_found": 0,
      "phase_after_clean": {
        "enabled": true,
        "platforms": [
          "bugbot"
        ],
        "started": true,
        "net_new_blocker": false,
        "blocking_platform": ""
      },
      "small_findings_only": false,
      "small_findings_stop": false,
      "small_findings_rounds": 0,
      "small_findings_required_rounds": 2,
      "small_findings_paths": []
    }
  ]
}
```
</details>

File: scripts/development-workflow/pr-review-loop.sh [hunk 1]
```diff
@@ -8678,7 +8695,7 @@ for index in "${!platforms[@]}"; do
       fi
       aggregate_result="clean"
       ;;
-    needs_fixes|escalate)
+    needs_fixes|waiting_on_reviewer|escalate)
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply scripts/development-workflow/pr-review-loop.sh]: Fixed in 6d75f483: `waiting_on_reviewer` now has its own aggregate arm, so it remains a terminal non-clean hold without setting ready-phase net-new blocker telemetry. Added `waiting_on_reviewer_not_folded_into_blocker_arm` to prevent regression.

**Comment by lhpaul** [comment]
bugbot run
